### PR TITLE
Add CalcJoinPoolShares and CalcExitPoolCoinsFromShares pool queries to Stargate whitelist

### PR DESCRIPTION
**Pull request by georgemc98**
Related to #2956 and dependent on PR #2972  

## What is the purpose of the change

This PR adds the `CalcJoinPoolShares` and `CalcExitPoolCoinsFromShares` queries added in PR #2972 to the whitelisted Stargate queries in the wasmbindings.

## Brief Changelog

*(for example:)*
 
  - Add CalcJoinPoolShares query to whitelisted Stargate queries
  - Add CalcExitPoolCoinsFromShares query to whitelisted Stargate queries

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? yes
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? yes
  - How is the feature or change documented? not documented

### CODE REVIEW

**Comment by p0mvn** [comment]
Thanks @georgemc98 . Can you please rebase / merge `main` here?

**Comment by georgemc98** [comment]
> Thanks @georgemc98 . Can you please rebase / merge `main` here?

Yes, just pushed my commits.

**Comment by georgemc98** [comment]
> Thanks @georgemc98 . Can you please rebase / merge `main` here?

@p0mvn Is there anything else you'd like me to do for this PR?

**Comment by georgemc98** [comment]
@p0mvn Hi Roman, this is a blocker for us.  Do you have an update on when we can get this reviewed?

**Comment by georgemc98** [comment]
Dependent on this PR: https://github.com/osmosis-labs/osmosis/pull/3242

**Comment by p0mvn** [comment]
Why is this PR so large? Is https://github.com/osmosis-labs/osmosis/pull/3242 included here as well?

Can we only keep the whitelist in this PR?

@georgemc98  I'm good with merging everything once comments are addressed and CI passes.

Please note that we are planning a v13 upgrade tomorrow. If this is not in by then, we will have to wait until v14 for this to be available in production.

Also, @ValarDragon mentioned that it should be possible to simulate these calculations off-chain precisely. I will follow up with him and get back on the approach.

**Comment by georgemc98** [comment]
> Why is this PR so large? Is #3242 included here as well?
> 
> Can we only keep the whitelist in this PR?
> 
> @georgemc98 I'm good with merging everything once comments in #3242 are addressed and CI passes.
> 
> Please note that we are planning a v13 upgrade tomorrow. If this is not in by then, we will have to wait until v14 for this to be available in production.
> 
> Also, @ValarDragon mentioned that it should be possible to simulate these calculations off-chain precisely. I will follow up with him and get back on the approach.

I'm fixing the suggestions in #3242 right now

**Comment by georgemc98** [comment]
@p0mvn What action is required to get the `required labels / state_compatability_labels` action to pass?

File: .github/workflows/mutest-issue-generate.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
         name: Get today's date
         id: date
         run: |
-          echo "::set-output name=today::$(date "+%Y/%m/%d")"
+          echo "name=today::$(date "+%Y/%m/%d")" >> $GITHUB_OUTPUT
```
Comment (p0mvn) [inline .github/workflows/mutest-issue-generate.yml:42]: Is this an accidental change?

Comment (georgemc98) [reply .github/workflows/mutest-issue-generate.yml:42]: Yes, no idea why this happened

File: x/gamm/keeper/export_test.go [hunk 2]
```diff
@@ -19,3 +19,5 @@ func (k Keeper) SetPool(ctx sdk.Context, pool types.PoolI) error {
 func (k Keeper) GetNextPoolIdAndIncrement(ctx sdk.Context) uint64 {
 	return k.getNextPoolIdAndIncrement(ctx)
 }
+
+//
```
Comment (p0mvn) [inline x/gamm/keeper/export_test.go:23]:
```suggestion
```

**Comment by georgemc98** [comment]
@p0mvn I changed the workflow files to match what is in `main`. This PR should be good to go now

File: .github/workflows/required_labels.yml [hunk 3]
```diff
@@ -9,10 +9,10 @@ jobs:
   state_compatability_labels:
     runs-on: ubuntu-latest
     steps:
-      - uses: mheap/github-action-required-labels@v2
+      - uses: RusAkh/github-action-required-labels
```
Comment (p0mvn) [inline .github/workflows/required_labels.yml:12]: Should this also be reverted?

Comment (georgemc98) [reply .github/workflows/required_labels.yml:12]: Yes

**Comment by georgemc98** [comment]
@p0mvn Is there anything else I should do to get this merged in?

**Review by p0mvn — APPROVED** [review]

**Review by czarcas7ic — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -47,6 +47,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * [#2788](https://github.com/osmosis-labs/osmosis/pull/2788) Add logarithm base 2 implementation.
 * [#2739](https://github.com/osmosis-labs/osmosis/pull/2739),[#3356](https://github.com/osmosis-labs/osmosis/pull/3356) Add pool type query, and add it to stargate whitelist
 * [#2956](https://github.com/osmosis-labs/osmosis/issues/2956) Add queries for calculating amount of shares/tokens you get by providing X tokens/shares when entering/exiting a pool
+* [#3217](https://github.com/osmosis-labs/osmosis/pull/3217) Add three pool queries to the registered Stargate queries list.
```
Comment (mattverse) [inline CHANGELOG.md:50]:
```suggestion
* [#3217](https://github.com/osmosis-labs/osmosis/pull/3217) Add `CalcJoinPoolShares`, `CalcExitPoolCoinsFromShares`, `CalcJoinPoolNoSwapShares` to the registered Stargate queries list.
```

**Review by mattverse — APPROVED** [review]
LGTM

**Pull request merged**